Cannibalism Update:tm:

WenlockTheBritishHobo opened the pull request:
## What Does This PR Do

Adds 16 limb-based dishes, as well as a a sprite variant for kebabs if one uses a bone rod instead of a metal rod. Also adds the bone rod and a recipe for the bone rod. Additionally, adds a new reagent: Lead.

As most of the recipes are just <species> limb in cooking device, I'll list the dish and cooking device first:

- Moffolo Wings (Nian): Deep Fryer
- Bug Bar (Kidan): Cutting Board
- Calamari (Drask): Deep Fryer
- Feline Mignon (Tajaran): Frying Pan (Stove). There is also an optional step to add 5u Wine.
- Frankfurrter (Vulp): Grill
- Frog Leg (Skrell): Frying Pan (Stove)
- Grey Gruel (Grey): Cooking Pot (Stove)
- Hiss Kebab (Unathi): Grill
- Leg of Ham (Human): Oven
- Plasma-Bone Broth (Plasmaman): Cooking Pot (Stove)
- Pork Rind (Human): Deep Fryer
- Tree Salad (Diona): Prep Bowl
- Turkish Delight (Slime People): Cutting Board
- Wafers (IPC): Cutting Board (Also, Wafers contains Lead)

Moffolo Wild Wings and Bug Burger are very similar to Moffolo Wings and Bug Bar, respectively.
Moffolo Wild Wings is the same except you add 5u BBQ Sauce. 
Bug Burger is a Bun and a Bug Bar on a Cutting Board.

The kebab bone rod variants are the same recipes except you use a bone rod instead of a metal rod. Bone Rods can be made from bone sheets, obtainable from using the bone dropped from Lavaland fauna.

## Why It's Good For The Game

I can eat a Vox's limbs, why not the rest of them?
Also, this means more Chef content, additionally promotes a bit of interaction between Mining and the Chef.

## Images of changes

<img width="147" height="257" alt="image" src="https://github.com/user-attachments/assets/6585d211-b181-4c97-9e2d-5e5804bb7de2" />
<img width="584" height="401" alt="image" src="https://github.com/user-attachments/assets/23259102-bc3d-48d6-9308-ae6ecc510c6d" />

## Testing

Made each limb-based dish by: spawning in the required species and cutting off one of their limbs, then spawning in any additional ingredients if the recipe called for it, then carrying out the recipe.

Ensured I could make bone_rods with bone sheets, ensured each food items with new items for their "trash" actually spawned the relevant item, ensured the Plasma-Bone Broth could be burnt like plasma sheets.

Tasted each limb-based dish. Checked that both Pork Rinds and Calamari had RNG for their sprites when they were spawned in from making the relevant recipe.

Ensured Lead actually gave lead poisoning (70% chance brain damage, different 70% chance for kidney damage but does toxin if there's either no kidneys or the kidneys are dead).

Made sure crew and Ian (Corgis) can wear the empty Moffolo Wild Wings bucket (both regular and wild moffolo wings use the same item for their trash).

## Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
<img width="831" height="96" alt="image" src="https://github.com/user-attachments/assets/052be12e-41c4-4075-8c4d-2f77e8f730a8" />

## Changelog

:cl:
add: Added new limb-based dishes for the rest of the species, so Vox aren't lonely.
add: Added bone rod variants of Kebab dishes
add: New reagent Lead
imageadd: Added icons for the new dishes
/:cl:

WenlockTheBritishHobo committed:
dmi time

WenlockTheBritishHobo committed:
.dmi stuff done, maybe

WenlockTheBritishHobo committed:
Created cannibalism.dm and moved KFV's obj stuff.

WenlockTheBritishHobo committed:
Update cannibalism.dm

WenlockTheBritishHobo committed:
Wild Wings and Bug Bar

WenlockTheBritishHobo committed:
A few more.

WenlockTheBritishHobo committed:
Meow Mein

WenlockTheBritishHobo committed:
Lead and Plasma-Bone Broth

WenlockTheBritishHobo committed:
Multiple possible sprites.

WenlockTheBritishHobo committed:
Last food item. Next recipes and the Bone Rod.

WenlockTheBritishHobo committed:
Adding bone rod sprites and sprites for bone rod variants of food with metal rods in them

WenlockTheBritishHobo committed:
Fix complaining of the compiler.

WenlockTheBritishHobo committed:
Bone Rods

WenlockTheBritishHobo committed:
Almost everything done.

WenlockTheBritishHobo committed:
Bone Rods ticked.

WenlockTheBritishHobo committed:
dme updated on compile.

WenlockTheBritishHobo committed:
Forgor the BBQ variant of Moffolo Wings needed a unique path.

WenlockTheBritishHobo committed:
Probably make Plasma-Bone Broth flammable like plasma sheets.

WenlockTheBritishHobo committed:
Should light now, will test later.

WenlockTheBritishHobo committed:
Some recipes, and processor.

WenlockTheBritishHobo committed:
Processor was complaining, also changed a parameter to be readable.

WenlockTheBritishHobo committed:
More bone rod.

WenlockTheBritishHobo committed:
The rest of them.

WenlockTheBritishHobo committed:
No more errors on compile.

WenlockTheBritishHobo committed:
Recipes tested and things fixed.

WenlockTheBritishHobo committed:
Lead works.

WenlockTheBritishHobo committed:
Kidneys dead? Toxin.

WenlockTheBritishHobo committed:
Soy Sauce in the Meow Mein.

WenlockTheBritishHobo committed:
Working sprite RNG.

WenlockTheBritishHobo commented:
!wiki_label

WenlockTheBritishHobo committed:
Merge remote-tracking branch 'upstream/master' into Cannibalism_Update

WenlockTheBritishHobo commented:
> Purely a personnal point of view: I think it would make more sense to craft bone rods from using a knife or sharp object on a bone in order to carve it rather than making a sheet and then using said sheet to make the rod(s). (Might give it a 1 or 2 seconds do-after ?) Unless of course you or someone else plans to use bone sheets for something else later (Or did they already exist? I have a hard time differenciating which lines are added and which are not on phone)
> 
> Also could you (if you choose to keep the bone sheets, which would be perfectly fine) add an image showing the bone sheet to the PR please ? Unless they already existed

Bone sheets are the bones dropped from butchered Lavaland fauna already. As for requiring the use of a tool for the bone rods, I could do that.

WenlockTheBritishHobo committed:
Make Linters happy

CRUNCH-Borg commented on code/modules/food_and_drinks/food/foods/cannibalism.dm:
```diff
@@ -0,0 +1,196 @@
+		log_and_set_aflame(user, I)
+	return TRUE
+
+/obj/item/food/plasmabone_broth/attackby__legacy__attackchain(obj/item/I, mob/living/user, params)
```
Food is on the new attack chain. Use `item_interaction()` instead.

Gatchapod commented:
The way I see it, this is detrimental to the entire roleplay environment.
It promotes unhinged behaviours like robbing the morgue or very specifically growing specimens for that purpose. But soylent green and KFV, you may say...

Soylent green is a classic reference and can be played around by denying its actual nature as part of the very same reference. It is genuinely amusing.
KFV, a mistake in my book, can still be reasonably explained as Vox bodies literally doubling as a food source on arkships. If you are a creation of genius bioengineers, it is not outside of the realm of possibility that you are a little too multipurpose for your own safety. KFV is also fine from the mechanic standpoint as there is always a soft limit on the amount of you produce. Vox bodies cannot be acquired through cargo nor hydroponics so every limb you get is a valuable rarity. Many of the dishes above can be mass-produced.

This is just jarring as you will likely have a fully-functional hydroponics ready to supply an entire system with food from a single tiny room and all the food is processed into wolpin cubes because today we eat Vulps.

So yes, despite all the good will and amazing work put into this PR, I continue to believe it will be a net negative on the server. Past the initial shock value it will almost definitely end up being just something used to harass people who literally just want to maintain a shred of sanity IC.
If I see kitchen being filled with limbless corpses and an assortment of these meals, I'm probably just going to cryo because at this point we pretend that we live in dystopian corpo-hell with overpriced food vendors perpetually in our sights while simultaneously living in a clearly post-scarcity period where food can be grown at mind-bending speed while also experiencing famine-induced necrophagia. No, I'm out at this point, I don't know what to even how to do barebones roleplay anymore.

Ahzpy reviewed (approved):
Sprites LGTM

DGamerL reviewed (changes requested):
Gotta fix the linters first
Code seemed fine at a glance

CRUNCH-Borg commented on code/modules/food_and_drinks/food/foods/cannibalism.dm:
```diff
@@ -0,0 +1,196 @@
+	if(I.get_heat())
+		log_and_set_aflame(user, I)
+	else
+		return ..()
```
```suggestion
/obj/item/food/plasmabone_broth/item_interaction(mob/living/user, obj/item/used, list/modifiers)
	if(!used.get_heat())
		return ..()
	log_and_set_aflame(user, used)
		return ITEM_INTERACT_COMPLETE
```
This will fix it.

WenlockTheBritishHobo committed:
Apply suggestions from CRUNCH's code review to fix Plasma-Bone Broth's flame.

Co-authored-by: CRUNCH <143041327+CRUNCH-Borg@users.noreply.github.com>
Signed-off-by: Wenlock_The_British_Hobo <168675688+WenlockTheBritishHobo@users.noreply.github.com>

WenlockTheBritishHobo committed:
Merge remote-tracking branch 'upstream/master' into Cannibalism_Update

WenlockTheBritishHobo committed:
Make linters happy

PollardTheDragon commented on code/modules/cooking/recipes/midway_recipes.dm:
```diff
@@ -63,6 +63,77 @@
+
+	return PCWJ_CHECK_INVALID
+
+/datum/cooking/recipe/mw
```
Can these names be more descriptive?

WenlockTheBritishHobo committed:
Updated bone rod description to be less vague.

Co-authored-by: PollardTheDragon <144391971+PollardTheDragon@users.noreply.github.com>
Signed-off-by: Wenlock_The_British_Hobo <168675688+WenlockTheBritishHobo@users.noreply.github.com>

WenlockTheBritishHobo committed:
More descriptive recipe names in code

WenlockTheBritishHobo committed:
Replace Meow Mein sprites with Feline Mignon

I'll see about using the Meow Mein sprite for chef craftable Chow Mein, as the issue is with Meow Mein not the sprite.

WenlockTheBritishHobo committed:
Change code from Meow Mein to Feline Mignon

WenlockTheBritishHobo committed:
Merge remote-tracking branch 'upstream/master' into Cannibalism_Update

DGamerL commented on code/game/objects/items/stacks/bone_rods.dm:
```diff
@@ -0,0 +1,35 @@
+GLOBAL_LIST_INIT(bone_rod_recipes, list (
+
+	))
```
Empty bone rod recipe list?

WenlockTheBritishHobo replied:
CRUNCH intends to add crafting recipes with them in a future PR, though now I realise I don't need to put an empty list if I'm not adding any crafting recipes myself.

CRUNCH-Borg replied:
> CRUNCH intends to add crafting recipes with them in a future PR, though now I realise I don't need to put an empty list if I'm not adding any crafting recipes myself.

Yeah, no need for that now. The list can be added when there's recipes to populate it.

DGamerL commented on code/modules/reagents/chemistry/reagents/toxins.dm:
```diff
@@ -123,6 +123,28 @@
+			if(our_kidneys && our_kidneys.damage < 60)
+				our_kidneys.receive_damage(1, TRUE)
+			else
+				M.adjustToxLoss(1)
```
So only kill kidneys if they're already almost down? Why not the other way around?

WenlockTheBritishHobo replied:
It only kills kidneys if kidneys aren't dead, so from 0-59 health given 60 health is when they're dead.

WenlockTheBritishHobo committed:
Don't need the blank recipe list in this PR.

DGamerL commented on code/game/objects/items/stacks/bone_rods.dm:
```diff
@@ -0,0 +1,30 @@
+	amount = 50
+
+/obj/item/stack/bone_rods/Initialize(mapload, new_amount, merge)
+	. = ..()
```
```suggestion
```

DGamerL committed:
Update code/game/objects/items/stacks/bone_rods.dm

Signed-off-by: DGamerL <108773801+DGamerL@users.noreply.github.com>

DGamerL committed:
Update code/game/objects/items/stacks/bone_rods.dm

Signed-off-by: DGamerL <108773801+DGamerL@users.noreply.github.com>
